### ISSUE TITLE
DL3003 Use WORKDIR to switch directory false positive

### ISSUE THREAD

**Issue by pbrisbin** [hadolint/hadolint#422]
- [x] This is a bug report
- [ ] This is a feature request
- [x] I searched existing issues before opening this one

### Expected behavior

When compiling software in a layer, it is often required to change directory, build, change back, and remove the sources. This must happen all within one `RUN` to avoid the sources remaining in a layer in the image. Therefore, the `cd` must happen within the `RUN`.

A suggestion to use `WORKDIR` should not apply to this pattern and so the lint error is a false-positive.

### Actual behavior

Any `cd` within a `RUN` is flagged, including this pattern.

### Steps to reproduce the behavior

```dockerfile
RUN \
  git clone git://github.com/jpmens/jo.git /tmp/jo && \
  cd /tmp/jo && \
  autoreconf -i && \
  ./configure && \
  make check && \
  make install && \
  cd /tmp && \
  rm -rf ./jo

```

I tried to follow the _Rationale_ section of [the wiki](https://github.com/hadolint/hadolint/wiki/DL3003), which mentions a sub-shell:

```dockerfile
RUN \
  git clone git://github.com/jpmens/jo.git /tmp/jo && \
  (cd /tmp/jo && \
  autoreconf -i && \
  ./configure && \
  make check && \
  make install) && \
  rm -rf /tmp/jo
```

but the lint error is still flagged for this. Maybe I'm doing that wrong?

**Output of `hadolint --version` or
  `docker run --rm hadolint/hadolint hadolint --version`:**

```
Haskell Dockerfile Linter v1.17.6-0-gc918759
```

**Comment by meeech** [comment]
I would add DL3003 wiki page doesn't actually explain the rationale. The correct code example doesn't make use of `WORKDIR` command. There's no explanation as to why you shouldn't use cd.

**Comment by guillaume-d** [comment]
The inspiration for the rule probably comes from https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#workdir, where a longer, more convincing rationale can be found.

I also interpreted the subshell info as an escape hatch, but it did not work for me either.
Maybe the pattern `RUN ` [...] `(cd `[...]` && `[...]`) && rm -Rf `[...] could indeed be ignored?
